packages/browser-ppx/tests/at_platform.t

```diff
@@ -124,6 +124,71 @@ Pstr_value
   $ ./standalone.exe -impl input_let.ml -js | ocamlformat - --enable-outside-detected-project --impl
   let x = 42 [@@platform js]
 
+Pexp_let
+
+  $ cat > input_let.ml << EOF
+  > let x =
+  >   let y = 42 [@@platform js] in
+  >   let y = 44 [@@platform native] in
+  >   y
+  > EOF
+
+  $ ./standalone.exe -impl input_let.ml | ocamlformat - --enable-outside-detected-project --impl
+  let x =
+    let y = 44 [@@platform native] in
+    y
+
+  $ ./standalone.exe -impl input_let.ml -js | ocamlformat - --enable-outside-detected-project --impl
+  let x =
+    let y = 42 [@@platform js] in
+    y
+
+Pexp_let
+
+  $ cat > input_let.ml << EOF
+  > let x =
+  >   let y = 42 [@@platform js] in
+  >   let y = 44 [@@platform native] in
```

Comment:
This snapshot is indeed interesting because currently, the switch%platform makes more sense in those scenarios, right? 

I know that it's just a test, but trying to understand better how it works

Reply:
Yep, in case you want to handle a single variable as `y` the `switch%platform` is the way:
```reason
let y = switch%platform () {
  | Client => 42
  | Server => 44
}
```

But there will be cases which we just want to say that this code is "browser_only".
Example:
```reason
let make = (~initial, ~onClick as [@platform js] onClick) => {
  let (count, [@platform js] setCount) = RR.useStateValue(initial);

  [@platform js]
  let onClick = e => {
    setCount(count + 1);

    Js.log("This prints to the console");
    Js.log2("Printing count", count);

    onClick(e);
  };

  //...
```

In this case, for example, onClick is not required on serve, so we ca remove it from native code, `setCount` is also `browser_only` so we change it to `_` on native, same with `onClick`
I don't know if it's possible to have `switch%platform` on those cases, for example:
```reason
switch%platform () {
  | Client =>
    let onClick = e => {
      setCount(count + 1);

      Js.log("This prints to the console");
      Js.log2("Printing count", count);

      onClick(e);
    };
    ();
  | Server => ()
};

// Outputs the code bellow, which will not work:
{
  let onClick = e => {
    setCount(count + 1);
    Js.log("This prints to the console");
    Js.log2("Printing count", count);
    onClick(e);
  };

  ();
};
```

I think `%browser_only` would cover the function (will still have the warning) but not the tuple of even the prop.
The prop is a very common case. It is pretty normal to have onClick props on a CustomComponent, as shown in the code above.

Reply:
imo we should consolidate to either `platform js` or `browser_only`. Keeping both is a bit confusing to me. If we decide for one, we could deprecate the other in one version and remove it in the next.